app/controllers/api/account.php

```diff
@@ -280,6 +280,12 @@
 
         $protocol = $request->getProtocol();
         $callback = $protocol . '://' . $request->getHostname() . '/v1/account/sessions/oauth2/callback/' . $provider . '/' . $project->getId();
+        $providerEnabled = $project->getAttribute('authProviders', [])[$provider.'Enabled'] ?? false;
+        
+        if (!$providerEnabled) {
+            throw new Exception(Exception::PROJECT_PROVIDER_DISABLED, 'This provider is disabled. Please enable the provider from your ' . APP_NAME . ' console to continue.');
```

Comment:
Do we need to specify message? Cant we use default from errors.php?
If so, did you copy this from somewhere? Should we fix this elsewhere too?

Reply:
Copied from few lines below.